src/main/java/teamfour/tasc/ui/CommandBox.java

```diff
@@ -48,9 +60,67 @@ public void configure(ResultDisplay resultDisplay, Logic logic) {
     private void addToPlaceholder() {
         SplitPane.setResizableWithParent(placeHolderPane, false);
         placeHolderPane.getChildren().add(commandTextField);
+        wordList.setTranslateX(80);
+        placeHolderPane.getChildren().add(wordList);
+        commandTextField.textProperty().addListener((observable, oldValue, newValue) -> {
+            checkCurrentWord(newValue);
+        });
+        commandTextField.setOnKeyPressed(new EventHandler<KeyEvent>() {
+            public void handle(KeyEvent ke) {
+                if (ke.getCode().equals(KeyCode.UP) || ke.getCode().equals(KeyCode.DOWN)) {
+                    if (wordList.getItems().size() > 0) {
+                        wordList.getSelectionModel().select(0);
+                        wordList.requestFocus();
+                    }
+                }
+            }
+        });
+        wordList.setOnKeyPressed(new EventHandler<KeyEvent>() {
+            public void handle(KeyEvent ke) {
+                if (ke.getCode().equals(KeyCode.ENTER)) {
+                    autoComplete(wordList.getSelectionModel().getSelectedItem());
+                    commandTextField.requestFocus();
+                    commandTextField.selectPositionCaret(0);
+                    commandTextField.end();
+                }
+            }
+        });
         FxViewUtil.applyAnchorBoundaryParameters(commandPane, 0.0, 0.0, 0.0, 0.0);
         FxViewUtil.applyAnchorBoundaryParameters(commandTextField, 0.0, 0.0, 0.0, 0.0);
     }
+    
+    private void autoComplete(String word) {
+        String[] words = commandTextField.getText().split(" ");
+        words[words.length-1] = word + " ";
+        commandTextField.setText(StringUtils.join(words, " "));
+    }
+    
+    private void setWordChoices(String word) {
+        if (word.trim().equals("")) {
+            wordList.setItems(null);
+            wordList.setPrefHeight(0);
+            return;
+        }
+        String[] keywords = {"add", "by", "clear", "complete", "delete", "exit", "find", "from", "help", "hide", "list", "relocate", "repeat", "redo", "select", "show", "to", "undo", "update"};
+        ObservableList<String> words = FXCollections.observableArrayList();
+        int endIndex = word.length();
+        for(String keyword: keywords) {
+            if (keyword.length() >= endIndex) {
+                if (keyword.substring(0, endIndex).equalsIgnoreCase(word)) {
+                    words.add(keyword);
+                }
+            }
+        }
+        wordList.setItems(null);
+        wordList.setItems(words);
+        wordList.setPrefHeight(words.size() * 16 + 2);
```

Comment:
The magic numbers (16 and 2) could be made a static const int, easier to change since it could depend on font size, margin or something